Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt

```diff
@@ -55,10 +59,41 @@ class RATProfileCreateFragment : Fragment(R.layout.rat_profile_create_fragment),
             zipCodeInputEdit.doAfterTextChanged { viewModel.zipCodeChanged(it.toString()) }
 
             // Phone
-            phoneInputEdit.doAfterTextChanged { viewModel.phoneChanged(it.toString()) }
+            phoneInputEdit.doAfterTextChanged {
+                // Propagate phone number to view model if it matches the pattern
+                if (Patterns.PHONE.matcher(it.toString()).matches()) {
+                    viewModel.phoneChanged(it.toString())
+                } else {
+                    viewModel.phoneChanged("")
+                }
+            }
+            phoneInputEdit.setOnFocusChangeListener { _, hasFocus ->
+                // Validate phone number
+                if (!hasFocus && !Patterns.PHONE.matcher(phoneInputEdit.text.toString()).matches()) {
+                    phoneInputLayout.error = "Invalid phone number"
+                } else {
+                    phoneInputLayout.error = null
+                }
+            }
+            phoneInputEdit.addTextChangedListener(PhoneNumberFormattingTextWatcher())
 
             // E-mail
-            emailInputEdit.addEmojiFilter().doAfterTextChanged { viewModel.emailChanged(it.toString()) }
+            emailInputEdit.addEmojiFilter().doAfterTextChanged {
+                // Propagate email to view model if it matches the pattern
+                if (Patterns.EMAIL_ADDRESS.matcher(it.toString()).matches()) {
+                    viewModel.emailChanged(it.toString())
+                } else {
+                    viewModel.emailChanged("")
+                }
+            }
+            emailInputEdit.setOnFocusChangeListener { _, hasFocus ->
+                // Validate email
+                if (!hasFocus && !Patterns.EMAIL_ADDRESS.matcher(emailInputEdit.text.toString()).matches()) {
+                    emailInputLayout.error = "Invalid email address"
```

Comment:
should be localised ?

Reply:
@harambasicluka pinged UA in #3764

to not increase complexity of PRs, maybe merge this and do a follow up PR?

Reply:
Ok, should we add least `strings` keys ? Normally UA update or translate them , but don't create new ones

Reply:
No problem. Will add that real quick ...

Reply:
Should be fine now